chainer/distributions/bijector.py

```diff
@@ -0,0 +1,88 @@
+from chainer.functions.math import exponential
+from chainer.functions.math import identity
+
+
+class Bijector():
+
+    """Interface of Bijector.
+
+    `Bijector` is implementation of bijective (invertible) function that is
+    used by `TransformedDistribution`. The three method `_forward`, `_inv` and
+    `_logdet_jac` have to be defined in inhereted class.
+    """
+
+    def __init__(self):
+        self.cache_x_y = (None, None)
```

Comment:
This caching mechanism seems a trade-off between speed and memory.  Users may want to turn off this feature.
This issue can be addressed in another PR.

Reply:
Wait.  There's a more significant issue.

In the current implementation of `TransformedDistribution.cdf`, we cannot compute the following.
```
d = TransformedDistribution(...)
y = d.sample(...)
q = d.cdf(y)
g, = chainer.grad(outputs=(q,), inputs=(y,))
```

Reply:
Oh, no...
Caching does not retain computational graph between input and output. But in many cases, `inv`  calculation is heavy and computational graph between input and output is needless.
Is it good solution that we add option to using cache? or is there any good solution that we can calculate gradient and cache?